src/components/policy/policy_external/src/policy_table/types.cc

```diff
@@ -1559,24 +1592,32 @@ ConsentRecords::~ConsentRecords() {}
 ConsentRecords::ConsentRecords(const Json::Value* value__)
     : CompositeType(InitHelper(value__, &Json::Value::isObject))
     , consent_groups(impl::ValueMember(value__, "consent_groups"))
+    , external_consent_status_groups(
+          impl::ValueMember(value__, "external_consent_status_groups"))
     , input(impl::ValueMember(value__, "input"))
     , time_stamp(impl::ValueMember(value__, "time_stamp")) {}
 
 Json::Value ConsentRecords::ToJsonValue() const {
   Json::Value result__(Json::objectValue);
   impl::WriteJsonField("consent_groups", consent_groups, &result__);
+  impl::WriteJsonField("external_consent_status_groups",
+                       external_consent_status_groups,
+                       &result__);
   impl::WriteJsonField("input", input, &result__);
   impl::WriteJsonField("time_stamp", time_stamp, &result__);
   return result__;
 }
 
 bool ConsentRecords::is_valid() const {
   if (struct_empty()) {
-    return initialization_state__ == kInitialized && Validate();
+    return initialization_state__ == kUninitialized && Validate();
```

Comment:
@LevchenkoS Honestly my initial reaction to this change was like yours, since this change is merged and slightly modified from already implemented functionality lets ask it's initial author to give you a thorough explanation, which I am currently is unable to provide in deep. @dev-gh, please comment.